File: server/src/SmartMap/Control/ServerFeedbackException.php
<?php

namespace SmartMap\Control;


/**
 * A ServerFeedbackException is used to give the client feedback that is displayable to the user,
 * but that should ideally never happen (that's why it is an exception).
 *
 * @author Pamoi
 */
class ServerFeedbackException extends \Exception
{
    function __construct($message = "", $code = 0, \Exception $previous = null)
    {
        parent::__construct($message, $code, $previous);
    }
}

File: SmartMap/src/ch/epfl/smartmap/servercom/ServerFeedbackException.java
/**
 * 
 */
package ch.epfl.smartmap.servercom;

/**
 * This exception contains an error message that can be shown to the user.
 * It is sent back by the server but should ideally never happen.
 * 
 * @author Pamoi
 */
public class ServerFeedbackException extends SmartMapClientException {

    private static final long serialVersionUID = 1L;

    public ServerFeedbackException() {
        super();
    }

    public ServerFeedbackException(String message) {
        super(message);
    }

    public ServerFeedbackException(Throwable throwable) {
        super(throwable);
    }
}
